sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/AnsiUtil.scala

```diff
@@ -48,8 +47,12 @@ object AnsiUtil extends Arm {
         s"NaN, Infinity or out-of-range values. To return NULL instead, use 'try_cast'. " +
         s"If necessary set ${SQLConf.ANSI_ENABLED.key} to false to bypass this error."
 
-    def throwDateTimeException: Unit = {
-      throw new DateTimeException(msg)
+    // These are the arguments required by SparkDateTimeException class to create error message.
+    val errorClass= "CAST_INVALID_INPUT"
+    val messageParameters = Array("Nan/Infinity", "DOUBLE", "TIMESTAMP", SQLConf.ANSI_ENABLED.key)
+
+    def throwSparkDateTimeException: Unit = {
```

Comment:
any reason not to make the Nan/Infinity a parameter here so that at least we can tell the user which one of those it is?

Reply:
> We need a separate message for when the input column is a float32.

Spark's error message is "DOUBLE" for both float and Double type. So keeping it as it is.